### PR TITLE
DOC-849 Add benchmark processor to Self-Managed

### PR DESCRIPTION
**Pull request by asimms41**
## Description

Resolves [DOC-849](https://redpandadata.atlassian.net/browse/DOC-849)
Review deadline: 11th December

Related PR to add the same content to Cloud docs: https://github.com/redpanda-data/cloud-docs/pull/151

## Page previews

[`benchmark` processor](https://deploy-preview-125--redpanda-connect.netlify.app/redpanda-connect/components/processors/benchmark/)

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

[DOC-849]: https://redpandadata.atlassian.net/browse/DOC-849?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rockwotj — APPROVED** [review]

File: modules/components/pages/processors/benchmark.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  interval: 5s
+  count_bytes: true
+```
+
+== Throughput statistics
+
+This output logs the following rolling statistics at a <<interval,configurable interval>> to help you to understand the current performance of your pipeline:
+
+- The number of messages processed per second.
+- The number of bytes of messages processed per second (optional).
```
Comment (ooesili) [inline modules/components/pages/processors/benchmark.adoc:28]: "bytes of messages" feels weird to me here. I would just say "bytes processed per second"

**Review by ooesili — APPROVED** [review]

File: modules/components/pages/processors/benchmark.adoc (hunk 1 above)
Comment (asimms41) [reply modules/components/pages/processors/benchmark.adoc]: Updated.

File: modules/components/pages/processors/benchmark.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
+= benchmark
+// tag::single-source[]
+:type: processor
+:page-beta: true
+:categories: ["Utility"]
+
+component_type_dropdown::[]
+
+Logs throughput statistics for messages that pass through this processor, and provides a summary of those statistics over the lifetime of the processor.
```
Comment (Deflaimun) [inline modules/components/pages/processors/benchmark.adoc:9]:
```suggestion
Logs message throughput statistics for this processor and summarizes them over its lifetime.
```

messages thought this processor seems redudant. Consider this simplification

File: modules/components/pages/processors/benchmark.adoc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+- The number of messages processed per second.
+- The number of bytes processed per second (optional).
+
+For example:
+
+```bash
+INFO rolling stats: 1 msg/sec, 407 B/sec
+```
+
+When the processor shuts down, it also writes a summary of the number and size of messages processed during the lifetime of the processor to the logs. For example:
```
Comment (Deflaimun) [inline modules/components/pages/processors/benchmark.adoc:36]:
```suggestion
When the processor shuts down, it also logs a summary of the number and size of messages processed during its lifetime. For example:
```

**Review by Deflaimun — APPROVED** [review]
approved see comments

File: modules/components/pages/processors/benchmark.adoc (hunk 2 above)
Comment (asimms41) [reply modules/components/pages/processors/benchmark.adoc:9]: Generally, I'd avoid long noun strings. I've made some other updates. Thanks for the feedback.

File: modules/components/pages/processors/benchmark.adoc (hunk 3 above)
Comment (asimms41) [reply modules/components/pages/processors/benchmark.adoc:36]: Updated.

File: modules/components/pages/processors/benchmark.adoc (hunk 2 above)
Comment (Deflaimun) [reply modules/components/pages/processors/benchmark.adoc]: Nice! I like the update

File: modules/components/pages/processors/benchmark.adoc [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+=== `interval`
+
+How often to emit rolling statistics. Set to `0`, if you only want to log summary statistics when the processor shuts down.
+
+*Type*: `string`
+
+*Default*: `5s`
+
+=== `count_bytes`
```
[the remark this answers is not in the document]
Comment (ooesili) [reply modules/components/pages/processors/benchmark.adoc:52]: This processor has already been released with this name so we can't change it now without breaking the interface, but I will keep this in mind going forward.

**Pull request merged**